src/databricks/labs/ucx/assessment/azure.py

```diff
@@ -129,33 +126,53 @@ def _safe_get_cluster_policy(self, policy_id: str) -> Policy | None:
     def _get_azure_spn_from_config(self, config: dict) -> set[AzureServicePrincipalInfo]:
         """Detect azure service principals from a dictionary of spark configs
         This checks for the existence of an application id by matching fs.azure.account.oauth2.client.id
-        For each application id identified, retrieve the relevant storage account, and corresponding tenant id
+        For each application id identified, retrieve the relevant storage account, client secret and tenant id
         """
         set_service_principals = set[AzureServicePrincipalInfo]()
-        spn_application_id, secret_scope, secret_key, tenant_id, storage_account = None, "", "", "", ""
         matching_spn_app_id_keys = [key for key in config.keys() if "fs.azure.account.oauth2.client.id" in key]
         for spn_app_id_key in matching_spn_app_id_keys:
             # retrieve application id of spn
-            spn_application_id, secret_scope, secret_key = self._get_key_from_config(spn_app_id_key, config)
+            spn_application_id = self._get_key_from_config(spn_app_id_key, config)
             if spn_application_id is None:
                 continue
 
+            storage_account, secret_scope, secret_key, tenant_id = None, None, None, None
+            tenant_key = "fs.azure.account.oauth2.client.endpoint"
+            client_secret_key = "fs.azure.account.oauth2.client.secret"
+
             # retrieve storage account configured with this spn
             storage_account_matched = re.search(STORAGE_ACCOUNT_EXTRACT_PATTERN, spn_app_id_key)
             if storage_account_matched:
                 storage_account = storage_account_matched.group(1).strip(".")
-                tenant_key = "fs.azure.account.oauth2.client.endpoint." + storage_account
+                # adjust the key to lookup for tenant id & secret id
+                tenant_key = tenant_key + f".{storage_account}"
+                client_secret_key = client_secret_key + f".{storage_account}"
+
+            # retrieve client secret of spn
+            matching_secret_keys = [key for key in config.keys() if re.search(client_secret_key, key)]
+            if len(matching_secret_keys) == 0 or matching_secret_keys[0] is None:
+                secret_scope, secret_key = None, None
             else:
-                tenant_key = "fs.azure.account.oauth2.client.endpoint"
+                client_secret = self._get_key_from_config(matching_secret_keys[0], config, False)
+                if client_secret is not None:
+                    secret_matched = re.findall(SECRET_PATTERN, client_secret)
+                    if len(secret_matched) == 0:
+                        logger.warning('Secret in config stored in plaintext.')
+                    else:
+                        secret_scope, secret_key = secret_matched[0][0], secret_matched[0][1]
```

Comment:
Nesting gets too deep, split this method up

Reply:
done